src/store/selectors/decals.js

```diff
@@ -28,10 +21,13 @@ export const selectDecalById = (state, { decalId } = {}) => {
 
 
 export const selectDecalsByUserId = createSelector(
-  [selectDecals, getUserId],
-  (decals, userId) => {
-    return Object.values(decals).filter((decal) => {
-      return decal?.attributes?.userId === userId
-    })
+  [selectDecals, selectUserById],
+  (decals, user) => {
+    if (user) {
+      return user.relationships.decals?.data.map(({ id }) => {
+        return decals[id]
+      })
+    }
+    return []
```

Comment:
We should prefer to return `undefined` if no user is found since there is a difference between there being no decals for the user and no user at all.